.github/workflows/go.yml

```diff
@@ -183,13 +192,13 @@ jobs:
         run: ci/scripts/go_build.sh $(pwd)
       - name: Test
         shell: bash
-        run: ci/scripts/go_test.sh $(pwd)
+        run: ci/scripts/go_test.sh $(pwd)      
```

Comment:
nit
```suggestion
        run: ci/scripts/go_test.sh $(pwd)
```

Reply:
ran trim trailing whitespace to fix that :)

And yea, tested it on my fork before I removed the runs on PRs to have it only run on pushes to apache/arrow repo.